client/App.jsx

```diff
@@ -18,98 +46,50 @@ const Container = styled.div`
   padding: 5px;
 `;
 
+function mostHelpful(reviews) {
+  return reviews.reduce((best, current) => {
+    const helpfulness = (review) => (review.meta.helpful
+      / (review.meta.helpful + review.meta.unhelpful));
+
+    if (helpfulness(current) > helpfulness(best)) {
+      return current;
+    }
+
+    return best;
+  });
+}
+
 class App extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
       currentGame: 1,
-      reviews: [{
-        _id: '0',
-        gameId: 0,
-        date: `${new Date()}`,
-        overall: 1,
-        title: '',
-        review: '',
-        recommend: false,
-        nickname: '',
-        location: '',
-        email: '',
-        buyForSelf: false,
-        ageBracket: 1,
-        gender: 1,
-        graphics: 1,
-        gameplay: 1,
-        appeal: 1,
-        ownershipBracket: 1,
-        purchaseOnline: false,
-        readReviews: false,
-        recommendBGS: 1,
-        meta: {
-          helpful: 1,
-          unhelpful: 1,
-        },
-      }],
-      mostFavorable: {
-        _id: '0',
-        gameId: 0,
-        date: `${new Date()}`,
-        overall: 1,
-        title: '',
-        review: '',
-        recommend: false,
-        nickname: '',
-        location: '',
-        email: '',
-        buyForSelf: false,
-        ageBracket: 1,
-        gender: 1,
-        graphics: 1,
-        gameplay: 1,
-        appeal: 1,
-        ownershipBracket: 1,
-        purchaseOnline: false,
-        readReviews: false,
-        recommendBGS: 1,
-        meta: {
-          helpful: 1,
-          unhelpful: 1,
-        },
-      },
-      mostUnfavorable: {
-        _id: '0',
-        gameId: 0,
-        date: `${new Date()}`,
-        overall: 1,
-        title: '',
-        review: '',
-        recommend: false,
-        nickname: '',
-        location: '',
-        email: '',
-        buyForSelf: false,
-        ageBracket: 1,
-        gender: 1,
-        graphics: 1,
-        gameplay: 1,
-        appeal: 1,
-        ownershipBracket: 1,
-        purchaseOnline: false,
-        readReviews: false,
-        recommendBGS: 1,
-        meta: {
-          helpful: 1,
-          unhelpful: 1,
-        },
-      },
+      reviews: [dummyReview],
+      mostFavorable: dummyReview,
+      mostCritical: dummyReview,
       ratingCounts: [0, 0, 0, 0, 0],
       ratingAverages: [1, 1, 1, 1],
       totalReviews: 0,
+      filterMatrix: [
```

Comment:
what does `filterMatrix` do?

Reply:
This was kinda what I was worried about, and why I picked this PR.  If you have something like this where it's really not obvious what it does and how, is that a problem you solve by writing documentation, by adding comments, or by thinking about how to rewrite the code so it's more obvious what it's doing? (or maybe some other strategy?)

To answer the question, it's an array where each member corresponds to a possible filter on the data. Each "filter" includes a boolean for whether it's in use, a truth test, and an informative text string to display when it's being applied. 

When I was writing it I considered making a factory function or class and building this array during ComponentDidMount; would that have been clearer?